fail2ban/client/fail2bancmdline.py

```diff
@@ -0,0 +1,279 @@
+
+	def initCmdLine(self, argv):
+		verbose = 1
+		try:
+			# First time?
+			initial = (self._argv is None)
+
+			# Command line options
+			self._argv = argv
+			logSys.info("Using start params %s", argv[1:])
+
+			# Reads the command line options.
+			try:
+				cmdOpts = 'hc:s:p:xfbdviqV'
+				cmdLongOpts = ['loglevel=', 'logtarget=', 'syslogsocket=', 'async', 'timeout=', 'help', 'version']
+				optList, self._args = getopt.getopt(self._argv[1:], cmdOpts, cmdLongOpts)
+			except getopt.GetoptError:
+				self.dispUsage()
+				return False
+
+			ret = self.__getCmdLineOptions(optList)
+			if ret is not None:
+				return ret
+
+			logSys.debug("-- conf: %r, args: %r", self._conf, self._args)
+
+			if initial and PRODUCTION: # pragma: no cover - can't test
+				verbose = self._conf["verbose"]
+				if verbose <= 0:
+					logSys.setLevel(logging.ERROR)
+				elif verbose == 1:
+					logSys.setLevel(logging.WARNING)
+				elif verbose == 2:
+					logSys.setLevel(logging.INFO)
+				elif verbose == 3:
+					logSys.setLevel(logging.DEBUG)
+				else:
+					logSys.setLevel(logging.HEAVYDEBUG)
+				# Add the default logging handler to dump to stderr
+				logout = logging.StreamHandler(sys.stderr)
+				# set a format which is simpler for console use
+				formatter = logging.Formatter('%(levelname)-6s %(message)s')
+				# tell the handler to use this format
+				logout.setFormatter(formatter)
+				logSys.addHandler(logout)
+
+			# Set expected parameters (like socket, pidfile, etc) from configuration,
+			# if those not yet specified, in which read configuration only if needed here:
+			conf = None
+			for o in CONFIG_PARAMS:
+				if self._conf.get(o, None) is None:
+					if not conf:
+						self.configurator.readEarly()
+						conf = self.configurator.getEarlyOptions()
+					self._conf[o] = conf[o]
+
+			logSys.info("Using socket file %s", self._conf["socket"])
+
+			logSys.info("Using pid file %s, [%s] logging to %s",
+				self._conf["pidfile"], self._conf["loglevel"], self._conf["logtarget"])
+
+			if self._conf.get("dump", False):
+				ret, stream = self.readConfig()
+				self.dumpConfig(stream)
+				return ret
+
+			# Nothing to do here, process in client/server
+			return None
+		except Exception as e:
+			output("ERROR: %s" % (e,))
+			if verbose > 2:
+				logSys.exception(e)
+			return False
+
+	def readConfig(self, jail=None):
+		# Read the configuration
+		# TODO: get away from stew of return codes and exception
+		# handling -- handle via exceptions
+		stream = None
+		try:
+			self.configurator.Reload()
+			self.configurator.readAll()
+			ret = self.configurator.getOptions(jail)
+			self.configurator.convertToProtocol()
+			stream = self.configurator.getConfigStream()
+		except Exception, e:
+			logSys.error("Failed during configuration: %s" % e)
+			ret = False
+		return ret, stream
+
+	@staticmethod
+	def dumpConfig(cmd):
+		for c in cmd:
+			output(c)
+		return True
+
+	@staticmethod
+	def exit(code=0): # pragma: no cover - can't test
```

Comment:
Currently the global exit handler is overwritten in test cases (to test forked processes exit code), so it can't (we need a return code). 

Indeed, we could rewrite it using mock module, but the overhead is unnecessarily (several times larger) for coverage of this 2 lines.